### DISCUSSION TITLE
Feature request: typescript support

### DISCUSSION THREAD

**Discussion by styxlab** [maizzle/framework#1215] [General]
This is a feature request for rewriting the library in **typescript** as it gives a much better DX experience and helps in finding bugs at an earlier stage. This could be done progressively, starting with new functions.

Thanks so much for this amazing framework that covers a lot of the issue that one encounters when building html emails!

**Comment by cossssmin** [comment]
Hey sorry but will have to close this one for now as it's not a priority right now. We might get back to it in the future depending on how things evolve. Thanks!

**Comment by danawoodman** [comment]
Would be really nice to have TS support esp for users (like me) who are using it via the API. Would you consider reopening?

**Comment by chrisbbreuer** [comment]
I would also be interested in this. Just started implementing it into [Stacks](https://github.com/stacksjs/stacks) as the default "engine" to power the styling/templating of email notifications. Let's see where this goes as I am familiarizing myself with this codebase 😊

<img width="699" alt="Screenshot 2022-12-08 at 5 25 40 PM" src="https://user-images.githubusercontent.com/6228425/206505787-ce27184d-a6fa-48f8-940c-cfab21a1705e.png">

**Comment by flolu** [comment]
I would also be interested in TypeScript support

**Comment by shrbhara** [comment]
@chrisbbreuer - how did you solve this issue?

**Comment by flolu** [comment]
Any updates?

**Comment by joeyfigaro** [comment]
I'd also like to request TS support.

I'd be open to contributing / helping if this is something you are interested in working on.

**Comment by cossssmin** [comment]
Anything we can do to improve the type definitions that we export is more than welcome, but rewriting the entire codebase in TS is not something I want to pursue.

Working on v5 and will soon be able to publish a dev branch for it so if you think that's something you can help with that would be great! Going to convert this issue to a discussion so that we keep it surfaced 👍
